### PR TITLE
Email mailing list (addresses #32)

### PR DESCRIPTION
**Pull request by benwiley4000**
Automatically emails mailing list at express server start with updates re: any new blog posts. Blog posts are plain text, wrapped at 80 characters, and include the same relevant info at the top as an HTML blog post, as well as a permalink. The base url for blog posts can be adjusted from `config.json`.

**After this PR is merged, assuming it is, I'd be happy to handle https://github.com/redbrick/static-site/issues/31**

Steps to test module from command line:

```
node
> var emailNewPosts = require('emailNewPosts');
> emailNewPosts(function (err) { console.log(err); });
```
If you have any errors, they'll be printed out.

NOTE: The address used for SMTP "from" will be the same address used for SMTP auth.

Pre-reqs:

Create a file in the root directory called mailing_list. It's not json, it's a newline-separated list. Each line is an email address. If you don't include any email addresses you won't email anyone.
Create a file in the root directory called email_update_log. It can be blank - if it is, running emailNewPosts will send an email for every post since the beginning of time to every user on your list. After that, the top line will display the latest timestamp and only newer posts will be sent to users' email inboxes.

Tips:

If you want to send only one post to your mailing list, make the top line of email_update_log a date earlier than your latest post but later than all others. You can do this easily by hand with any date string recognized by JavaScript, e.g. "January 1, 2016."
If you want to exclude some emails in the list from sending, just change the '@' sign to something else and that address will be ignored.

### CODE REVIEW

**Comment by benwiley4000** [comment]
Oops! Forgot to do the 80 character word wrap, but included now.

File: emailNewPosts.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  });
+}
+
+function writeArrayToFile (filename, array, callback) {
+  fs.writeFile(filename, array.join('\n'), callback);
+}
+
+function sendEmail (emailData, address, callback) {
+  var mailOptions = {
+    from: 'Redbrick <' + config.email.auth.user + '>',
```
Comment (VoyTechnology) [inline emailNewPosts.js]: Would it not be more beneficial to put it into config?

Comment (benwiley4000) [inline emailNewPosts.js]: Oh do you mean the name "Redbrick" for `mailOptions`?

Comment (benwiley4000) [inline emailNewPosts.js]: I apologize - for some reason when I view this comment on my phone your quoted selection is much larger (starts at line 24) - hence my confusion. I can change that.

Comment (benwiley4000) [inline emailNewPosts.js]: My latest commit should resolve these concerns. :smiley:

File: emailNewPosts.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  });
+}
+
+function writeArrayToFile (filename, array, callback) {
+  fs.writeFile(filename, array.join('\n'), callback);
+}
+
+function sendEmail (emailData, address, callback) {
+  var mailOptions = {
+    from: config.mailername + ' <' + config.email.auth.user + '>',
```
Comment (butlerx) [inline emailNewPosts.js]: instead of using config.mailername it would probably be better to use the author from blog post.

Comment (benwiley4000) [inline emailNewPosts.js]: Cool.

**Comment by benwiley4000** [comment]
All previously indicated concerns should be resolved. Let me know if there's anything else.

**Comment by butlerx** [comment]
thanks for your help and all the changes

**Comment by benwiley4000** [comment]
One inefficiency I noticed too late is that `wrap80` is applied to the same text once for each email address times the number of messages rather than only once for each message. This can be solved easily by moving the invocation of `wrap80` to the return statement of `getEmailBody()`

**Comment by butlerx** [comment]
would you mind creating this as an issue

**Comment by benwiley4000** [comment]
If it doesn't add too much bloat I can definitely create and issue and
submit a PR.

**Comment by butlerx** [comment]
go ahead

**Pull request merged**